qiita_pet/handlers/api_proxy/sample_template.py

```diff
@@ -0,0 +1,269 @@
+from __future__ import division
+import warnings
+
+from os import remove
+from os.path import exists, join
+
+from natsort import natsorted
+# This is the only folder in qiita_pet that should import from outside
+# qiita_pet. The idea is this proxies the call and response dicts we expect
+# from the Qiita API once we build it. This will be removed and replaced with
+#  API calls when the API is complete.
+from qiita_db.metadata_template.sample_template import SampleTemplate
+from qiita_db.study import Study
+from qiita_core.util import execute_as_transaction
+from qiita_core.qiita_settings import qiita_config
+
+from qiita_db.metadata_template.util import (load_template_to_dataframe,
+                                             looks_like_qiime_mapping_file)
+from qiita_db.util import get_mountpoint
+from qiita_ware.metadata_pipeline import (
+    create_templates_from_qiime_mapping_file)
+from qiita_pet.util import convert_text_html
+from qiita_pet.handlers.api_proxy.util import check_access
+
+
+def sample_template_summary_get_req(samp_id, user_id):
+    """Returns a summary of the sample template metadata columns
+
+    Parameters
+    ----------
+    samp_id : int
+        SampleTemplate id to get info for
+    user_id : str
+        User requesting the sample template info
+
+    Returns
+    -------
+    dict of list of tuples
+        Dictionary object where the keys are the metadata categories
+        and the values are list of tuples. Each tuple is an observed value in
+        the category and the number of times its seen.
+        Format {num_samples: value,
+                category: [(val1, count1), (val2, count2), ...], ...}
+    """
+    access_error = check_access(samp_id, user_id)
+    if access_error:
+        return access_error
+    template = SampleTemplate(int(samp_id))
+    df = template.to_dataframe()
+    out = {'num_samples': df.shape[0],
+           'summary': {}}
+
+    # drop the samp_id column if it exists
+    if 'study_id' in df.columns:
+        df.drop('study_id', axis=1, inplace=True)
+    cols = list(df.columns)
+    for column in cols:
+        counts = df[column].value_counts()
+        out['summary'][str(column)] = [(str(key), counts[key])
+                                       for key in natsorted(counts.index)]
+
+    return out
+
+
+@execute_as_transaction
```

Comment:
The ones that make change to the DB and filesystem are. The only one that isn't is the get request, since it only does select statements.